### PR TITLE
Bug + Refactor: Features-> Auth Module .

### PR DESCRIPTION
**Pull request by Darkeye14**
Fixes #Issue_Number
MIFOSAC-277

1)Fixes Post Login navigation issue.
2) CleanUp of Auth Module
3)Applying Spotless and detekt
Please Add Screenshots If there are any UI changes.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [ ] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [ ] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by Darkeye14** [comment]
https://github.com/user-attachments/assets/ac6e9f19-0bbd-4bef-8f4e-fd41a87e46de

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/navigation/PasscodeNavGraph.kt [hunk 1]
```diff
@@ -23,20 +26,33 @@ internal fun NavGraphBuilder.passcodeNavGraph(navController: NavHostController)
...
+            saveState = false
+        }
+        launchSingleTop = true
+        restoreState = false
+    }
+    navigate(MifosNavGraph.MAIN_GRAPH , options)
+
+
+
+}
```
Comment (itsPronay) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/navigation/PasscodeNavGraph.kt:58]: we don't need it. 
The error can be fixed by changing the Navroute of `MifosNavGraph.MainGraph`.

Comment (Darkeye14) [reply mifosng-android/src/main/java/com/mifos/mifosxdroid/navigation/PasscodeNavGraph.kt]: Done Bhai

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/AndroidClient.kt [hunk 2]
```diff
@@ -30,11 +34,14 @@ fun AndroidClient() {
...
         authNavGraph(
-            navigatePasscode = {},
+            navigatePasscode = navController::navigateToPasscodeScreen,
             navigateHome = navController::navigateHome,
             updateServerConfig = {}
         )
 
+        composable(MifosNavGraph.MAIN_GRAPH){
+            HomeNavigation()
+        }
```
Comment (itsPronay) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/AndroidClient.kt:44]: why are we adding this here? we already have one for `homeGraph`

File: mifosng-android/src/main/res/navigation/nav_graph.xml [hunk 3]
```diff
@@ -74,6 +74,39 @@
...
+            app:destination="@id/individualCollectionSheetDetailsFragment" />
+
+    </fragment>
+
+
+    <fragment
+        android:id="@+id/generateCollectionSheetFragment"
+        android:name="com.mifos.mifosxdroid.online.generatecollectionsheet.GenerateCollectionSheetFragment"
+        android:label="GenerateCollectionSheetFragment" />
+
```
Comment (itsPronay) [inline mifosng-android/src/main/res/navigation/nav_graph.xml:109]: Why are we adding this again? Wasn't it already removed during the previous cleanup process?

File: mifosng-android/src/main/res/navigation/nav_graph.xml [hunk 4]
```diff
@@ -110,6 +143,47 @@
...
+            android:name="loanAndClientName"
+            app:argType="com.mifos.core.objects.collectionsheet.LoanAndClientName" />
+        <argument
+            android:name="paymentTypeOptions"
+            app:argType="com.mifos.core.objects.accounts.loan.PaymentTypeOptions[]" />
+        <argument
+            android:name="clientId"
+            app:argType="integer" />
+    </fragment>
+
```
Comment (itsPronay) [inline mifosng-android/src/main/res/navigation/nav_graph.xml:186]: same here as well

**Pull request merged**